ehealthid-rp/src/main/java/com/oviva/ehealthid/relyingparty/ws/AuthEndpoint.java

```diff
@@ -45,17 +47,23 @@ public class AuthEndpoint {
 
   private final Pages pages = new Pages(new TemplateRenderer());
 
+  private final Counter authRequests;
+  private final Counter authSuccesses;
+
   public AuthEndpoint(
       URI baseUri,
       RelyingPartyConfig relyingPartyConfig,
       SessionRepo sessionRepo,
       TokenIssuer tokenIssuer,
-      AuthenticationFlow authenticationFlow) {
+      AuthenticationFlow authenticationFlow,
+      PrometheusMeterRegistry prometheusMeterRegistry) {
     this.baseUri = baseUri;
     this.relyingPartyConfig = relyingPartyConfig;
     this.sessionRepo = sessionRepo;
     this.tokenIssuer = tokenIssuer;
     this.authenticationFlow = authenticationFlow;
+    this.authRequests = prometheusMeterRegistry.counter("auth.requests");
```

Comment:
I am not 100% sure I did what you are looking for with this, or if you would rather want something like counters within the actual flow components.

Also, I decided for the smallest implementation for now, regarding actually emitted metrics. We could track failures as well, instead of calculating `total-success`, which, yes, equals `failures+in_flight`, but for that, I would like a bit more input on how we want to do that.

We could, for example, increment a failure counter in every failure branch, just plain, or we could also attach a reason to that counter, differentiating why things went belly up.
As with all things, pros and cons, happy to discuss.

Reply:
Let's start with the current implementation and then go from there.
Most likely you are also deploying a reverse proxy/load balancer infront and have metrics for that. Based on the status codes that should also give the error-rate.